### PR TITLE
[BEAM-1340] Adds __all__ tags to modules in package apache_beam/io

### PR DESCRIPTION
**Pull request by chamikaramj**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @robertwb

File: sdks/python/apache_beam/coders/typecoders.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ def MakeXyzs(v):
 from apache_beam.coders import coders
 from apache_beam.typehints import typehints
 
+__all__ = ['CoderRegistry']
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/typecoders.py]: Actually, I think registry is the only public thing here. But remove this; I'll take care of it with my coder changes.

File: sdks/python/apache_beam/io/concat_source.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
 
 from apache_beam.io import iobase
 
+__all__ = ['ConcatSource', 'ConcatRangeTracker']
```
Comment (robertwb) [inline sdks/python/apache_beam/io/concat_source.py]: Do we want users using this? (Maybe...if so should we mark as experimental?)

File: sdks/python/apache_beam/io/fileio.py [hunk 3]
```diff
@@ -37,6 +37,8 @@
 
 DEFAULT_SHARD_NAME_TEMPLATE = '-SSSSS-of-NNNNN'
 
+__all__ = ['FileSink']
```
Comment (robertwb) [inline sdks/python/apache_beam/io/fileio.py]: Why do we have FileBasedSource but FileSink? Make an alias (if it's too late to rename) and export exactly one version.

File: sdks/python/apache_beam/io/filesystem.py [hunk 4]
```diff
@@ -30,6 +30,9 @@
 
 DEFAULT_READ_BUFFER_SIZE = 16 * 1024 * 1024
 
+__all__ = ['CompressionTypes', 'CompressedFile', 'FileMetadata', 'FileSystem',
+           'BeamIOError', 'MatchResult']
```
Comment (robertwb) [inline sdks/python/apache_beam/io/filesystem.py]: Let's not export BeamIOError. We can still throw it.

File: sdks/python/apache_beam/io/range_trackers.py [hunk 5]
```diff
@@ -24,6 +24,10 @@
 
 from apache_beam.io import iobase
 
+__all__ = ['OffsetRangeTracker', 'GroupedShuffleRangeTracker',
```
Comment (robertwb) [inline sdks/python/apache_beam/io/range_trackers.py]: GroupedShuffleRangeTracker should not be public. (Probably shouldn't even be in this file...)

File: sdks/python/apache_beam/coders/typecoders.py (hunk 1 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/coders/typecoders.py]: Reverted.

File: sdks/python/apache_beam/io/concat_source.py (hunk 2 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/concat_source.py]: Reverted. Added a comments about backwards compatability.

File: sdks/python/apache_beam/io/fileio.py (hunk 3 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/fileio.py]: Done.

File: sdks/python/apache_beam/io/filesystem.py (hunk 4 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filesystem.py]: Done.

File: sdks/python/apache_beam/coders/typecoders.py (hunk 1 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/coders/typecoders.py]: Reverted.

File: sdks/python/apache_beam/io/concat_source.py (hunk 2 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/concat_source.py]: Reverted. Added a comment about backwards compatibility.

File: sdks/python/apache_beam/io/fileio.py (hunk 3 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/fileio.py]: Done.

File: sdks/python/apache_beam/io/filesystem.py (hunk 4 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filesystem.py]: Done.

File: sdks/python/apache_beam/io/range_trackers.py (hunk 5 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/range_trackers.py]: Done.

**Pull request closed without merging**